ui/components/multichain/multi-srp/srp-list/srp-list.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+import { KeyringMetadata, KeyringObject } from '@metamask/keyring-controller';
+import Card from '../../../ui/card';
+import {
+  Box,
+  IconName,
+  Icon,
+  Text,
+  IconSize,
+} from '../../../component-library';
+import {
+  JustifyContent,
+  Display,
+  TextColor,
+  FlexDirection,
+  AlignItems,
+  BlockSize,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import {
+  getMetaMaskAccounts,
+  getMetaMaskHdKeyrings,
+} from '../../../../selectors/selectors';
+import { InternalAccountWithBalance } from '../../../../selectors/selectors.types';
+import { SrpListItem } from './srp-list-item';
+
+type KeyringObjectWithMetadata = KeyringObject & { metadata: KeyringMetadata };
+
+export const SrpList = ({
+  onActionComplete,
+  hideShowAccounts,
+}: {
+  onActionComplete: (id: string) => void;
+  hideShowAccounts?: boolean;
+}) => {
+  const hdKeyrings: KeyringObjectWithMetadata[] = useSelector(
+    getMetaMaskHdKeyrings,
+  );
+  // This selector will return accounts with nonEVM balances as well.
+  const accountsWithBalances: Record<string, InternalAccountWithBalance> =
+    useSelector(getMetaMaskAccounts);
+
+  const showAccountsInitState = useMemo(
+    () =>
+      hdKeyrings.reduce(
+        (acc: Record<string, boolean>, _, index) => ({
+          ...acc,
+          [index]: Boolean(hideShowAccounts), // if hideShowAccounts is true, show all accounts by default
+        }),
+        {},
+      ),
```

Comment:
Instead of a reduce, couldn't we use `new Array(hdKeyrings.length).fill(hideShowAccounts)`? I'm pretty sure we used that pattern elsewhere for some other multi-srp components?